conjure-java-core/src/integrationInput/java/com/palantir/product/ExternalLongExample.java

```diff
@@ -0,0 +1,224 @@
+package com.palantir.product;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.palantir.conjure.java.lib.internal.ConjureCollections;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import javax.annotation.Generated;
+
+@JsonDeserialize(builder = ExternalLongExample.Builder.class)
+@Generated("com.palantir.conjure.java.types.BeanGenerator")
+public final class ExternalLongExample {
+    private final long externalLong;
+
+    private final Optional<Long> optionalExternalLong;
+
+    private final List<Long> listExternalLong;
+
+    private volatile int memoizedHashCode;
+
+    private ExternalLongExample(
+            long externalLong, Optional<Long> optionalExternalLong, List<Long> listExternalLong) {
+        validateFields(optionalExternalLong, listExternalLong);
+        this.externalLong = externalLong;
+        this.optionalExternalLong = optionalExternalLong;
+        this.listExternalLong = Collections.unmodifiableList(listExternalLong);
+    }
+
+    @JsonProperty("externalLong")
+    public long getExternalLong() {
+        return this.externalLong;
+    }
+
+    @JsonProperty("optionalExternalLong")
+    public Optional<Long> getOptionalExternalLong() {
```

Comment:
@cakofony do you envisage wanting to change this to be an OptionalLong at some point? (I know we do this for conjure-native integers already in OptionalIntegerExample.java).

Or is that codepath something you'd put on the gotham fork?

Reply:
Yes, we do that in certain versions of this generator already, however I think that may be a bit too special for external type imports, and is likely a better fit for an integer64 type once that is implemented.